coffeecard/CoffeeCard.Library/Services/v2/MenuItemService.cs

```diff
@@ -0,0 +1,103 @@
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Threading.Tasks;
+using CoffeeCard.Common.Errors;
+using CoffeeCard.Library.Persistence;
+using CoffeeCard.Models.DataTransferObjects.v2.Product;
+using CoffeeCard.Models.DataTransferObjects.v2.Products;
+using CoffeeCard.Models.Entities;
+using Microsoft.EntityFrameworkCore;
+using Serilog;
+
+namespace CoffeeCard.Library.Services.v2
+{
+    public sealed class MenuItemService : IMenuItemService
+    {
+        private readonly CoffeeCardContext _context;
+
+        public MenuItemService(CoffeeCardContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<IEnumerable<MenuItemResponse>> GetAllMenuItemsAsync()
+        {
+            return await _context.MenuItems
+                .OrderBy(p => p.Id)
+                .Select(p => new MenuItemResponse
+                {
+                    Id = p.Id,
+                    Name = p.Name
+                })
+                .ToListAsync();
+        }
+
+
+        private async Task<bool> CheckMenuItemUniquenessAsync(string name, int id = 0)
+        {
+            var menuItem = await _context.MenuItems
+                .FirstOrDefaultAsync(p => p.Name == name && p.Id != id);
+
+            return menuItem == null;
+        }
+
+        public async Task<MenuItemResponse> AddMenuItem(AddMenuItemRequest newMenuItem)
+        {
+            var unique = await CheckMenuItemUniquenessAsync(newMenuItem.Name);
+            if (!unique)
+            {
+                throw new ConflictException($"Menu item already exists with name {newMenuItem.Name}");
+            }
+
+            var menuItem = new MenuItem()
+            {
+                Name = newMenuItem.Name
+            };
+
+            _context.MenuItems.Add(menuItem);
+            await _context.SaveChangesAsync();
+
+            var result = new MenuItemResponse
+            {
+                Id = menuItem.Id,
+                Name = menuItem.Name
+            };
+
+            return result;
+        }
+
+        public async Task<ChangedMenuItemResponse> UpdateMenuItem(UpdateMenuItemRequest changedMenuItem)
```

Comment:
I would suggest to change the order of operations in this method
1. Check that Id exists
2. Check for uniqueness of new name
3. Perform changes